include/LightGBM/config.h

```diff
@@ -76,126 +78,125 @@ struct Config {
   // desc = number of threads for LightGBM
-  // desc=for the best speed,set this to the number of **real CPU cores**,
-  // not the number of threads(most CPU using `hyper-threading`_ to generate 2 threads per CPU core)
-  // desc=do not set it too large if your dataset is small (do not use 64 threads for a dataset with 10,000 rows for instance)
-  // desc=be aware a task manager or any similar CPU monitoring tool might report cores not being fully utilized. **This is normal**
-  // desc=for parallel learning,should not use full CPU cores since this will cause poor performance for the network
+  // desc = ``0`` means default number of threads in OpenMP
+  // desc = for the best speed, set this to the number of **real CPU cores**, not the number of threads (most CPUs use `hyper-threading <https://en.wikipedia.org/wiki/Hyper-threading>`__ to generate 2 threads per CPU core)
+  // desc = do not set it too large if your dataset is small (for instance, do not use 64 threads for a dataset with 10,000 rows)
+  // desc = be aware a task manager or any similar CPU monitoring tool might report that cores not being fully utilized. **This is normal**
+  // desc = for parallel learning, do not use all CPU cores because this will cause poor performance for the network communication
   int num_threads = 0;
 
   // [doc-only]
-  // options=cpu,gpu
-  // desc = choose device for the tree learning, you can use GPU to achieve the faster learning
-  // desc=**Note**: it is recommended to use the smaller max_bin (e.g. 63) to get the better speed up
-  // desc=**Note**: for the faster speed,GPU use 32-bit float point to sum up by default,may affect the accuracy for some tasks.
-  // desc=You can set gpu_use_dp = true to enable 64 - bit float point, but it will slow down the training
-  // desc=**Note**: refer to `Installation Guide <./Installation-Guide.rst#build-gpu-version>`__ to build with GPU
+  // type = enum
+  // options = cpu, gpu
+  // desc = device for the tree learning, you can use GPU to achieve the faster learning
+  // desc = **Note**: it is recommended to use the smaller ``max_bin`` (e.g. 63) to get the better speed up
+  // desc = **Note**: for the faster speed, GPU uses 32-bit float point to sum up by default, so this may affect the accuracy for some tasks. You can set ``gpu_use_dp=true`` to enable 64-bit float point, but it will slow down the training
+  // desc = **Note**: refer to `Installation Guide <./Installation-Guide.rst#build-gpu-version>`__ to build LightGBM with GPU support
   std::string device_type = "cpu";
 
   // [doc-only]
-  // alias=random_seed
-  // desc=Use this seed to generate seeds for others, e.g. data_random_seed.
-  // desc=Will be override if set other seeds as well
-  // default=none
+  // alias = random_seed
+  // desc = this seed is used to generate other seeds, e.g. ``data_random_seed``, ``feature_fraction_seed``
+  // desc = will be overridden, if you set other seeds
   int seed = 0;
```

Comment:
Why default=none?

Reply:
I think it is a typo.

Reply:
I had the same thoughts, so deleted it.

What is your opinion about it https://github.com/Microsoft/LightGBM/pull/1409#pullrequestreview-125173261 ?

Reply:
Rolled back in #1872.